### PR TITLE
Rewrote quickstart based on docs best practices

### PR DESCRIPTION
**Pull request by melinath**
The change has a few parts to it:

1. Merged former setup & generate-providers pages into a single "quickstart"
2. Renamed Getting started -> Get started
3. Moved content that didn't fit in the quickstart (reference material related to make commands and container-based environments) to a reference page
4. Made some tweaks to the doctor script to check for additional requirements

See go/tpg-contribution-docplan for context

```release-note:none

```

Quickstart page:

localhost_1313_magic-modules_get-started_generate-providers_

Make command reference page:

localhost_1313_magic-modules_reference_make-commands_

### CODE REVIEW

File: docs/content/get-started/generate-providers.md [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+# Generate `google` and `google-beta` providers
+
+This quickstart guides you through setting up your development environment, making a change to `magic-modules`, generating provider changes to the `google` and `google-beta` Terraform providers, and running tests related to the change.
+
+## Before you begin
+
+1. [Install the gcloud CLI.](https://cloud.google.com/sdk/docs/install)
+1. In the Google Cloud console, on the project selector page, select or [create a Google Cloud project](https://cloud.google.com/resource-manager/docs/creating-managing-projects).
+   {{< hint info >}}
+   **Note:** If you don't already have a project to use for testing changes to the Terraform providers, create a project instead of selecting an existing poject. After you finish making your changes, you can delete the project, removing all resources associated with the project.
```
Comment (zli82016) [inline docs/content/get-started/generate-providers.md:21]: `removing` should be `remove` ?

File: docs/content/reference/make-commands.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+make provider VERSION=ga OUTPUT_PATH="$GOPATH/src/github.com/hashicorp/terraform-provider-google" PRODUCT=doesnotexist
+```
+
+#### Arguments
+
+- `OUTPUT_PATH`: Required. The location you are generating provider code into.
+- `VERSION`: Required. The version of the provider you are building into. Valid values are `ga` and `beta`.
+- `PRODUCT`: Limits generations to the specified folder within `mmv1/products` or `tpgtools/api`. Handwritten files from `mmv1/third_party/terraform` are always generated into the downstream regardless of this setting, so you can provide a non-existant product name to generate only handwritten code. Required if `RESOURCE` is specified.
+- `RESOURCE`: Limits generation to the specified resource within a particular product. For `mmv1` resources, matches the resource's `name` field (set in its configuration file).For `tpgtools` resources, matches the terraform resource name.
+- `ENGINE`: Modifies make provider to only generate code using the specified engine. Valid values are `mmv1` or `tpgtools`. (Providing `tpgtools` will still generate any prerequisite mmv1 files required for tpgtools.)
```
Comment (zli82016) [inline docs/content/reference/make-commands.md:39]: Should 'make provider' be  quoted with  backquotes ?

File: docs/content/get-started/generate-providers.md [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+     description: |
+       UPDATED_DESCRIPTION
+   ```
+1. Clone the `google` and `google-beta` provider repositories with the following commands:
+
+   ```bash
+   git clone https://github.com/hashicorp/terraform-provider-google.git $GOPATH/src/github.com/hashicorp/terraform-provider-google
+   git clone https://github.com/hashicorp/terraform-provider-google-beta.git $GOPATH/src/github.com/hashicorp/terraform-provider-google-beta
+   ```
+1. Generate changes for the `google` provider
```
Comment (zli82016) [inline docs/content/get-started/generate-providers.md:91]: Should add the link to the section `make commands` here?  I feel the section `make commands` is helpful to generate the providers especially for one product.

File: docs/content/get-started/generate-providers.md (hunk 1 above)
Comment (melinath) [reply docs/content/get-started/generate-providers.md:21]: I believe `removing` is correct - this is standard language used by quickstarts. See https://cloud.google.com/compute/docs/create-linux-vm-instance for example.

File: docs/content/get-started/generate-providers.md (hunk 3 above)
Comment (melinath) [reply docs/content/get-started/generate-providers.md:91]: I'm trying to limit the number of links that might disrupt a user's flow working through the quickstart - see https://cloud.google.com/guides/docs/quickstart for more details. The "`make` commands" reference is linked at the end of the doc for users who want to know more.

File: docs/content/reference/make-commands.md (hunk 2 above)
Comment (melinath) [reply docs/content/reference/make-commands.md]: Resolved in 6dc57b6b4

Comment (melinath) [reply docs/content/reference/make-commands.md]: Thanks!

File: docs/content/get-started/generate-providers.md (hunk 1 above)
Comment (zli82016) [reply docs/content/get-started/generate-providers.md:21]: I see. Thanks.

File: docs/content/get-started/generate-providers.md (hunk 3 above)
Comment (zli82016) [reply docs/content/get-started/generate-providers.md:91]: Got it. Thanks.

**Review by zli82016 — APPROVED** [review]
LGTM. The new quick start is much clearer.

**Comment by melinath** [comment]
Merging without VCR tests since there are no changes.

**Pull request merged**